src/plugins/vis_type_timeseries/server/routes/post_vis_schema.ts

```diff
@@ -17,59 +17,65 @@
  * under the License.
  */
 
-import Joi from 'joi';
-const stringOptionalNullable = Joi.string()
-  .allow('', null)
-  .optional();
-const stringRequired = Joi.string()
-  .allow('')
-  .required();
-const arrayNullable = Joi.array().allow(null);
-const numberIntegerOptional = Joi.number()
-  .integer()
-  .optional();
-const numberIntegerRequired = Joi.number()
-  .integer()
-  .required();
-const numberOptional = Joi.number().optional();
-const queryObject = Joi.object({
-  language: Joi.string().allow(''),
-  query: Joi.string().allow(''),
+import { schema } from '@kbn/config-schema';
+import { TypeOptions } from '@kbn/config-schema/target/types/types';
+
+const stringOptionalNullable = schema.nullable(schema.string());
+
+const stringRequired = schema.string();
+
+const arrayNullable = schema.arrayOf(schema.nullable(schema.any()));
+
+const validateInteger: TypeOptions<number>['validate'] = value => {
+  if (!Number.isInteger(value)) {
+    return `${value} is not an integer`;
+  }
+};
+const numberIntegerOptional = schema.maybe(schema.number({ validate: validateInteger }));
+const numberIntegerRequired = schema.number({ validate: validateInteger });
+
+const numberOptional = schema.maybe(schema.number());
+
+const queryObject = schema.object({
+  language: schema.string(),
+  query: schema.string(),
 });
-const stringOrNumberOptionalNullable = Joi.alternatives([stringOptionalNullable, numberOptional]);
-const numberOptionalOrEmptyString = Joi.alternatives(numberOptional, Joi.string().valid(''));
+const stringOrNumberOptionalNullable = schema.nullable(
+  schema.oneOf([stringOptionalNullable, numberOptional])
+);
+const numberOptionalOrEmptyString = schema.maybe(
+  schema.oneOf([numberOptional, schema.literal('')])
+);
 
-const annotationsItems = Joi.object({
+const annotationsItems = schema.object({
   color: stringOptionalNullable,
   fields: stringOptionalNullable,
-  hidden: Joi.boolean().optional(),
+  hidden: schema.maybe(schema.boolean()),
   icon: stringOptionalNullable,
   id: stringOptionalNullable,
   ignore_global_filters: numberIntegerOptional,
   ignore_panel_filters: numberIntegerOptional,
   index_pattern: stringOptionalNullable,
-  query_string: queryObject.optional(),
+  query_string: schema.maybe(queryObject),
   template: stringOptionalNullable,
   time_field: stringOptionalNullable,
 });
 
-const backgroundColorRulesItems = Joi.object({
-  value: Joi.number()
-    .allow(null)
-    .optional(),
+const backgroundColorRulesItems = schema.object({
+  value: schema.nullable(schema.number()),
```

Comment:
Looks like this needs a `schema.maybe` wrapper as well - in the Joi schema this was optional.